test/System/ClientSpec.hs

```diff
@@ -99,6 +99,27 @@ sys = parallel $ do
           unsub c1 sid
           takeMVar asyncWait1
 
+        it "receives others messages with headers" $ \(_, host, port) -> do
+          (c1, asyncWait1) <- testClient host port
+          (c2, asyncWait2) <- testClient host port
+
+          (lockAssure, assertAssure) <- asyncAssert (\_ -> return ())
```

Comment:
alter `asyncAssert` to be like the unit tests - takes a list of (Msg -> Expectation)

then we can assert the headers actually come through as expected

Reply:
pulled assertions into own package